packages/components/src/dropdown/test/index.js

```diff
@@ -1,98 +1,68 @@
 /**
  * External dependencies
  */
-import { fireEvent, render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 /**
  * Internal dependencies
  */
-import Dropdown from '../';
-
-function getButtonElement( container ) {
-	return container.querySelector( 'button' );
-}
-function getOpenCloseButton( container, selector ) {
-	return container.querySelector( selector );
-}
+import Dropdown from '..';
 
 describe( 'Dropdown', () => {
-	function expectPopoverVisible( container, value ) {
-		const popover = container.querySelector( '.components-popover' );
-		if ( value ) {
-			expect( popover ).toBeTruthy();
-		} else {
-			expect( popover ).toBeFalsy();
-		}
-	}
-
-	it( 'should toggle the dropdown properly', () => {
-		const expectButtonExpanded = ( container, expanded ) => {
-			expect( container.querySelectorAll( 'button' ) ).toHaveLength( 1 );
-			expect( getButtonElement( container ) ).toHaveAttribute(
-				'aria-expanded',
-				expanded.toString()
-			);
-		};
-
-		const {
-			container: { firstChild: dropdownContainer },
-		} = render(
+	it( 'should toggle the dropdown properly', async () => {
+		render(
 			<Dropdown
-				className="container"
-				contentClassName="content"
 				renderToggle={ ( { isOpen, onToggle } ) => (
 					<button aria-expanded={ isOpen } onClick={ onToggle }>
-						Toggleee
+						Toggle
 					</button>
 				) }
-				renderContent={ () => <span>test</span> }
+				renderContent={ () => 'Content' }
 			/>
 		);
 
-		expectButtonExpanded( dropdownContainer, false );
-		expectPopoverVisible( dropdownContainer, false );
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+		const button = screen.getByRole( 'button' );
 
-		const button = getButtonElement( dropdownContainer );
-		fireEvent.click( button );
+		expect( screen.queryByText( 'Content' ) ).not.toBeInTheDocument();
+		expect( button ).toBeInTheDocument();
```

Comment:
Just like https://github.com/WordPress/gutenberg/pull/45694#discussion_r1018562140, we should probably be using `toBeVisible()`.

```suggestion
		expect( button ).toBeVisible();
```

Reply:
Thanks. Needed to wrap them in `waitFor` because of the animation in `Popover`. See 87f8f79db31cf9036cc090b3f0634795de4ef746